### PR TITLE
fix(sklearn): raise ValueError in XGBRanker.score when qid is missing

### LINKED ISSUE #12334: [Bug] XGBRanker.score silently returns invalid metric values when qid is omitted
### Description
The `XGBRanker.score` method evaluates the ranking metric (e.g., NDCG) using the scikit-learn API. According to the docstring, it expects `X` to be a `pd.DataFrame` containing a special `qid` column because ranking evaluation requires query groups. 

However, if `X` is passed without the `qid` column (e.g., as a numpy array or a dataframe missing the special column), the method silently proceeds to create a `DMatrix` with `qid=None` and evaluates the metric. In the C++ backend, if `group_ptr` is empty, ranking metrics (like `rank_metric.cc`) fallback to treating the entire dataset as a single query/group. This results in `score` returning a completely meaningless and silently invalid metric value.

### Reproduction / Evidence

In `python-package/xgboost/sklearn.py`, the `score` method doesn't validate whether `qid` was actually extracted:

```python
    def score(self, X: ArrayLike, y: ArrayLike) -> float:
        ...
        X, qid = _get_qid(X, None)
        # If X has no 'qid' column, qid becomes None
        Xyq = DMatrix(
            X,
            y,
            qid=qid,
            missing=self.missing,
            enable_categorical=self.enable_categorical,
            nthread=self.n_jobs,
            feature_types=self.feature_types,
        )
        ...
        # Evaluates metric without query group boundaries if qid was None
        result_str = self.get_booster().eval(Xyq)
```

By contrast, the `XGBRanker.fit` method correctly validates this and raises an explicit error:
```python
        if kwargs.get("group", None) is None and qid is None:
            raise ValueError("Either `group` or `qid` is required for ranking task")
```

### Expected Behavior
`XGBRanker.score` should validate that `qid` is not `None` before creating the `DMatrix`, similar to what `fit` does. If `qid` is missing, it should raise a `ValueError` indicating that the `qid` column is required for ranking evaluation.

For example, appending a check inside `score`:
```python
        X, qid = _get_qid(X, None)
        if qid is None:
            raise ValueError(
                "For ranking task, `X` must be a DataFrame with a special `qid` "
                "column."
            )
```

### Environment
This affects the current master branch of the XGBoost Python package.

### PR DESCRIPTION
**Pull request by saitejabandaru-in**
Fixes #12334. If XGBRanker.score is evaluated without query IDs (qid column in X DataFrame), the C++ backend silently treats the entire evaluation dataset as a single query, producing invalid NDCG/MAP scores. This validation mirrors the checks done in fit().

### CODE REVIEW

**Comment by trivialfis** [comment]
Please run:
```
pre-commit run --from-ref master --to-ref HEAD
```
to fix the lint errors (or just fix it based on the error log from the CI).

**Comment by saitejabandaru-in** [comment]
Hi @trivialfis! Thanks for the review. I've formatted the `qid` validation check in `XGBRanker.score` to conform to black/flake8 standards and pushed the update.

**Review by trivialfis — APPROVED** [review]
